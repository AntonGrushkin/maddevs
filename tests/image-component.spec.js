import {
  mount
} from '@vue/test-utils';

import ImageComponent from '@/components/ui/image-component';

describe('Webp image component', () => {
  let wrapper;
  const props = {
    fileName: 'Marat',
    fileNameRetina: 'Marat@2x'
  };

  beforeEach(() => {
    wrapper = mount(ImageComponent, {
      propsData: props
    });
  });

  // ------ IMPORTANT ----- //
  test('is Vue\'s instance', () => {
    expect(wrapper.exists()).toBeTruthy();
  });

  test('renders correctly', () => {
    expect(wrapper.element).toMatchSnapshot();
  });
  // --------------------- //

  test('correctly props data', () => {
    expect(wrapper.props().fileName).toBe(props.fileName);
    expect(wrapper.props().fileNameRetina).toBe(props.fileNameRetina);
  });

  test('Has class if sectionIsRequired equal true', () => {
    expect(wrapper.find('picture')).toBeTruthy();
  });
});
