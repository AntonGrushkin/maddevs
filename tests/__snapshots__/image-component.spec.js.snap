// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Webp image component renders correctly 1`] = `
<picture>
  <source
    class="multi-image"
    loading="lazy"
    srcset="[object Object] ,[object Object] 2x"
    type="image/webp"
  />
   
  <img
    alt="Marat@2x"
    class="multi-image"
    loading="lazy"
    src="[object Object]"
    srcset="[object Object] 2x"
  />
</picture>
`;
